Tangem/App/Services/WalletConnect/WalletConnectV2Service/WalletConnectV2Service.swift

```diff
@@ -328,7 +328,11 @@ final class WalletConnectV2Service {
 
 extension WalletConnectV2Service: WalletConnectV2WalletModelProvider {
     func getModel(with address: String, in blockchain: BlockchainSdk.Blockchain) throws -> WalletModel {
-        guard let model = infoProvider.walletModels.first(where: { $0.wallet.blockchain == blockchain && $0.wallet.address == address }) else {
+        guard
+            let model = infoProvider.walletModels.first(where: {
+                $0.wallet.blockchain == blockchain && $0.wallet.address.caseInsensitiveCompare(address) == .orderedSame
+            })
+        else {
```

Comment:
```suggestion
let condition: (WalletModel) -> Bool = {
                $0.wallet.blockchain == blockchain && $0.wallet.address.caseInsensitiveCompare(address) == .orderedSame
            }
        guard let model = infoProvider.walletModels.first(where: condition) else {
```

может так кстати иногда нам писать? просто как предложение)

Reply:
ммм, не знаю, не уверен)) вроде саму структуру кода делает покрасивее, но сам код не упрощает и не оптимизирует, но и не ухудшает)) можно обсудить в группе

Reply:
Неа, потому что там берется информация по BlockchainNetwork и там вообще один блокчеин может быть за раз подключен. В 1 версии у нас хранится сессия, а в ней внутри лежит `BlockchainNetwork`, тут у нас нет доступа к сессии, т.к. она одна и мы можем понять только по адресу с какой деривацией нам надо взять модельку

Reply:
а сессии отличаются по топику, который хранится для каждой url свой, это ещё при создании соединения определяется в зависимости от дАппа. Тут же нам просто приходит сообщение: на подпиши это вот таким адресом О_о, а топик запроса присылают, а не сессии